File: schemas/member.py
from . import ma

class MemberSchema(ma.Schema):

    class Meta:
        strict = True

    id = ma.Int(dump_only=True)

    first_name = ma.Str()
    last_name = ma.Str()
    gender = ma.Str()

    email = ma.Str()

    joined_on = ma.Date(dump_only=True)

    access = ma.Int()
    sid = ma.Int()

    newsletter = ma.Bool()
    doing_it = ma.Bool()
    registered = ma.Bool()

    events_attended = ma.Nested('AttendanceSchema', many=True, exclude=('members_attended', 'member'), dump_only=True)

    ref = ma.URLFor('member', id='<id>')

File: src/api/members/resources/member.ts
import { BaseResource } from "utils/resource.base";
import { Gender } from "interfaces/member/gender";
import { Member } from "interfaces/member";

export class MemberResource extends BaseResource implements Member {

    protected readonly prefix: string = "/members"

    id: number;

    email?: string;

    firstName: string;
    
    lastName: string;
    
    gender: Gender;

    joinedOn: Date;

    access?: number;

    sid?: number;

    newsletter: boolean;

    doingIT: boolean;

    registered: boolean;

    getResourceName(): string {
        return "Member"
    }
}